rs/ledger_suite/icp/index/src/main.rs

```diff
@@ -492,7 +492,7 @@ fn get_block_range_from_stable_memory(
     start: u64,
     length: u64,
 ) -> Result<Vec<EncodedBlock>, String> {
-    let length = length.min(icp_ledger::max_blocks_per_request(&caller()) as u64);
+    let length = length.min(icp_ledger::max_blocks_per_request(&caller().into()) as u64);
```

Comment:
nit:
```suggestion
    let length = length.min(icp_ledger::max_blocks_per_request(&PrincipalId::from(caller())) as u64);
```
The same would apply on line 598 below.
(Although with this, you'd also need to add `use ic_base_types::PrincipalId;`.)